app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeScreen.kt

```diff
@@ -0,0 +1,241 @@
+import agency.five.codebase.android.movieapp.ui.home.mapper.HomeScreenMapper
+import agency.five.codebase.android.movieapp.ui.home.mapper.HomeScreenMapperImpl
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.Text
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import agency.five.codebase.android.movieapp.R
+import agency.five.codebase.android.movieapp.ui.component.*
+import androidx.compose.runtime.*
+
+val homeScreenMapper: HomeScreenMapper = HomeScreenMapperImpl()
+val movies = MoviesMock.getMoviesList()
+val popular = listOf(
+    MovieCategory.POPULAR_STREAMING,
+    MovieCategory.POPULAR_ON_TV,
+    MovieCategory.POPULAR_FOR_RENT,
+    MovieCategory.POPULAR_IN_THEATRES
+)
+val nowPlaying = listOf(
+    MovieCategory.NOW_PLAYING_MOVIES,
+    MovieCategory.NOW_PLAYING_TV
+)
+val upcoming = listOf(
+    MovieCategory.UPCOMING_TODAY,
+    MovieCategory.UPCOMING_THIS_WEEK
+)
+var popularCategoryViewState = homeScreenMapper.toHomeMovieCategoryViewState(
+    movies = movies,
+    movieCategories = popular, selectedMovieCategory = MovieCategory.POPULAR_STREAMING
+)
+var nowPlayingCategoryViewState = homeScreenMapper.toHomeMovieCategoryViewState(
+    movies = movies,
+    movieCategories = nowPlaying, selectedMovieCategory = MovieCategory.NOW_PLAYING_MOVIES
+)
+var upcomingCategoryViewState = homeScreenMapper.toHomeMovieCategoryViewState(
+    movies = movies,
+    movieCategories = upcoming, selectedMovieCategory = MovieCategory.UPCOMING_TODAY
+)
+
+
+@Composable
+fun HomeRoute(
+    onMovieCardClick: () -> Unit,
+    onFavoriteButtonClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val popularCategoryViewState by remember { mutableStateOf(popularCategoryViewState) }
+    val nowPlayingCategoryViewState by remember { mutableStateOf(nowPlayingCategoryViewState) }
+    val upcomingCategoryViewState by remember { mutableStateOf(upcomingCategoryViewState) }
+    HomeScreen(
+        popular = popularCategoryViewState,
+        nowPlaying = nowPlayingCategoryViewState,
+        upcoming = upcomingCategoryViewState,
+        onMovieCardClick = { onMovieCardClick() },
+        onFavoriteButtonClick = { onFavoriteButtonClick() },
+        modifier = modifier
+    )
+}
+
+@Composable
+fun HomeScreen(
+    popular: HomeMovieCategoryViewState,
+    nowPlaying: HomeMovieCategoryViewState,
+    upcoming: HomeMovieCategoryViewState,
+    onMovieCardClick: () -> Unit,
+    onFavoriteButtonClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+        LazyColumn(modifier = modifier.height(670.dp)) {
```

Comment:
reformat, wrong indentation

Reply:
shouldn't LazyColumn be fillMaxSize instead? what if you have a screen larger then 670 dps?